src/core/toga/window.py

```diff
@@ -257,3 +258,31 @@ def save_file_dialog(self, title, suggested_filename, file_types=None):
             The absolute path(str) to the selected location.
         """
         return self._impl.save_file_dialog(title, suggested_filename, file_types)
+
+    def open_file_dialog(self, title, initial_directory=None, file_types=None, multiselect=False):
```

Comment:
It is usually possible on some systems to include a text associated to the extensions, for instance

`Text Files (*.txt)` or `Awesome Text files (*.txt)`

So `file_types` could be expanded to include those cases?

Reply:
Sounds like a good idea to me. It might be easiest to create a light "datatype" class wrapper to capture these descriptions so we can use either a list or strings or a list of "datatype" objects.